### ISSUE TITLE
Use puppeteer to get the full page screenshot has a problem of duplication on both horizontal and vertical.

### ISSUE THREAD

**Issue by leosx** [cefsharp/CefSharp#2398]
**# Operating environment：** 
Operation System : Windows 10 Enterprise
Puppeteer version: 1.3.0
CefSharp version   : 66.0

In the CEF program, the Puppeteer full page screenshot is supported. The CEFSharp program has some problems with the use of puppeteer for the full page screenshot, and there is a problem of duplication on both horizontal and vertical.

**# Nodejs Code：** 
```
const fs = require('fs');
const net = require('net');
const puppeteer = require('puppeteer');
const request = require('request');
const PORT = 8088;

/**
 * get browser debugger URL
 */
async function GetDebuggerUrl() {
    const versionUrl = `http://127.0.0.1:${PORT}/json/version`;
    return new Promise((resolve, reject) => {
        request(versionUrl, function (error, response, body) {
            if (!error && response.statusCode == 200) {
                let entity = {};
                if (typeof (body) == 'string') {
                    entity = JSON.parse(body);
                    const url = entity.webSocketDebuggerUrl;
                    resolve(url);
                } else {
                    reject('error');
                }
            } else {
                reject('Get debugger Url faild！');
            }
        });
    });
}

(async () => {
    const dburl = await GetDebuggerUrl();
    let browser = await puppeteer.connect({ browserWSEndpoint: dburl, slowMo: 0 });
    let pages = await browser.pages();
    let pageOne = pages[0];
    await pageOne.goto("http://cnblogs.com");
    await pageOne.screenshot({ fullPage: true, path: 'test.png' });
    await browser.close();
})();
```

**Comment by amaitland** [comment]
> Does this problem also occur in the CEF Sample Application from
> nope!

What command line arguments did you run `cefclient` with? What version did you use?

I'd be surprised if this was a `CefSharp` bug.

**Comment by leosx** [comment]
cefclient version is : 3.3359.1774 
command line arguments is : cefclient.exe --remote-debugging-port=8088
nodejs code is not change.

**Comment by leosx** [comment]
download cef source code file name is : cef_binary_3.3359.1774.gd49d25f_windows64.rar

**Comment by leosx** [comment]
@amaitland where can see debug protocol code?

**Comment by amaitland** [comment]
65 branch is 3325

You should read the bug report template again, you need to use more command line arguments to compare to the WPF version.

**Comment by amaitland** [comment]
You can download the CEF and Chromium source for the Spotify build server

**Comment by leosx** [comment]
@amaitland we used branch 3.3325.1758 test. There is still this problem

**Comment by leosx** [comment]
@amaitland  we use cef of version 3.3325.1758 test. cef have not this problem,but cefsharp of version 65 has this problem.

**Comment by hfsime** [comment]
we also have this problem.we uesd chrome devtools protocal to test full screenshot in version 3325,it's ok.but we used cefsharp pre65 version,the screenshot only caputure the visible screen area,not the full page.

**Comment by leosx** [comment]
yes,if cefclient use this command line arguments. cef also has this problem.

I'm sorry we didn't read the ISSE_TEMPLATE.md file carefully. If I want remove --multi-threaded-message-loop command in CEFSharp? Which source code file can be modified to remove this command?

**Comment by leosx** [comment]
in WPF project of CefSharp setting code: 

CefSettings settings = new CefSettings();
settings.CefCommandLineArgs.Add("--remote-debugging-port", "8088");
settings.CefCommandLineArgs.Add("--multi-threaded-message-loop", "true");
Cef.Initialize(settings);

The problem still exists。

but if i use :  cefclient.exe --remote-debugging-port=8088 --multi-threaded-message-loop
start ceflient.exe ,don't have this problem. How to avoid this problem correctly?

**Comment by leosx** [comment]
@amaitland Is there any way to remove these two parameters so that they can be normal screenshots? I use cefclient to remove any of the parameters can be truncate to normal, but to remove any of the parameters in the cefsharp is not, or will be wrong, or the problem is still. thank you very mach!

**Comment by amaitland** [comment]
> To compare with WPF run cefclient --multi-threaded-message-loop --off-screen-rendering-enabled
> To compare with WinForms run cefclient --multi-threaded-message-loop

As per the issue template you are effectively comparing to the `WinForms` version.

**Comment by hfsime** [comment]
thanks! we used WinForms app instead of WPF app,it's ok. offscreen in  multi-threaded-message-loop scene caused this problem,we'll report this bug to cef.
